packages/oas-resolver/index.js

```diff
@@ -139,29 +140,53 @@ function testProtocol(input, backup) {
     return 'file:';
 }
 
-function resolveExternal(root, pointer, options, callback) {
-    var u = url.parse(options.source);
-    var base = options.source.split('\\').join('/').split('/');
+function getAbsoluteRefPath(source, $ref) {
+    let base = source.split('\\').join('/').split('/');
     let doc = base.pop(); // drop the actual filename
     if (!doc) base.pop(); // in case it ended with a /
+    base = base.join('/');
+    let u = url.parse(source);
+    let u2 = url.parse($ref);
+    let effectiveProtocol = testProtocol(u2.protocol, u.protocol)
+    if (effectiveProtocol === 'file:') {
+        return {
+            target: path.resolve(base ? base + '/' : '', $ref),
+            effectiveProtocol,
+            base
+        };
+    }
+    else {
+        return {
+            target: url.resolve(base ? base + '/' : '', $ref),
+            effectiveProtocol,
+            base
+        };
+    }
+}
+
+function resolveToName(ref, refs) {
```

Comment:
I know they haven't been using JsDocs, but it might be a good idea to introduce some on your new methods. In particular, this one is not so clear on the nature of the parameters and its purpose.